### ISSUE TITLE
No event hook to reposition element

### ISSUE THREAD

**Issue by Billy-** [nazar-pc/PickMeUp#111]
I know you can use an option to affect where the element is poisitioned **related to the triggered element**. I am trying to position the element dynamically with JS but all of the event hooks take place before the positioning is done, so any styles I add get overwritten.

I think we need an 'after_show' event hook.

Although given the scenario I am not sure what the difference between `'before_show'` and `'show'` is; considering they both take place before the element is shown.

It is important that you allow the user to do this as they may not want the widget to be positioned relative to the element that triggers it. That element could even be hidden. But the main issue I am having is on a mobile the widget gets positioned partly off the page. Would it also be an idea to add logic to check if the widget is going to positioned outside of the page, and if so, make some adjustments accordingly?

Regards

**Comment by nazar-pc** [comment]
You can always wrap datepicker in some `div` and apply custom positioning for it, while datepicker will just sit inside of it and never actually show/hide itself.

**Comment by Billy-** [comment]
Sorry I did not see your reply to this before. How would I wrap it as it's dynamically generated? Also would it not still try to position the datepicker with inline styles relative the the input that triggered it?

**Comment by nazar-pc** [comment]
If you place it into some `div`, there will be no direct connection with `input`, you'll need to establish that connection manually using available events.

**Comment by Billy-** [comment]
How can I place it in a `div`? The datepicker element is generated dynamically from JS?

``` JS
var $dateInpt = $('.js-date');

if($dateInpt.length) {
  $dateInpt.pickmeup({
    'hide_on_select': true
  });
}
```

**Comment by nazar-pc** [comment]
You have to basically using flat mode, not applying it to `input`. Like in demo: https://github.com/nazar-pc/PickMeUp/blob/master/js/demo.js#L2

**Comment by Billy-** [comment]
Hmm, don't you think it would be nice to have an event fire when the datepicker element is shown so that we can apply position dynamically without having to use the flat mode?

**Comment by Billy-** [comment]
Or allow the user to override the positioning function and pass their own function that positions / styles the element? I would be happy to help contribute to this feature

**Comment by nazar-pc** [comment]
I'm not sure I'd like to have it right now. The reason is that there are many people with multiple edge-cases and supporting literally all use-cases makes codebase too bloated.
I'm still collecting thoughts and hopefully one day I'll put my hands to fix all existing bugs, release new version and than starting refactoring that will lead to new major version with much better design and more possibilities for customization.

**Comment by Billy-** [comment]
Ok. Thanks for your help and I hope you add this feature in in the future - I do think it would be a very useful one. Let me know if you want any help! :)

**Comment by nazar-pc** [comment]
I've just published 3.1.0 that supports custom function in `position` option, so now it is possible to implement arbitrary positioning in generic way
